packages/devextreme/js/__internal/scheduler/m_utils_time_zone.ts

```diff
@@ -55,17 +57,66 @@ const getDaylightOffset = (startDate, endDate) => new Date(startDate).getTimezon
 
 const getDaylightOffsetInMs = (startDate, endDate) => getDaylightOffset(startDate, endDate) * toMs('minute');
 
-const calculateTimezoneByValue = (timezone, date = new Date()) => {
-  // NOTE: This check could be removed. We don't support numerical timezones
-  if (typeof timezone === 'string') {
-    const dateUtc = createUTCDate(date);
-    return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+const calculateTimezoneByValue = (timeZone, date = new Date()): number | undefined => {
+  if (!timeZone) {
+    return undefined;
+  }
+
+  let offset;
+  try {
+    offset = getStringOffset(timeZone, date);
+  } catch (e) {
+    return undefined;
+  }
+
+  if (offset === 'GMT') {
+    return 0;
   }
-  return timezone;
+
+  const isMinus = offset.substring(3, 4) === '-';
+  const hours = offset.substring(4, 6);
+  const minutes = offset.substring(7, 9);
+
+  const result = parseInt(hours, 10) + parseInt(minutes, 10) / 60;
```

Comment:
If something goes wrong and we receive the not a number string in ```hours``` or ```minutes``` variables, the ```parseInt``` will return the ```NaN``` value.
So, let's make this code a little more safe:
```suggestion
  const hoursStr = offset.substring(4, 6);
  const minutesStr = offset.substring(7, 9);
  const hours = parseInt(hoursStr, 10);
  const minutes = parseInt(minutesStr, 10);

  if (Number.isNaN(hours) || Number.isNaN(minutes)) {
    return undefined;
  }

  const result = (hours + minutes) / MINUTES_IN_HOUR;
```

And let's move the "magic number" ```60``` to the ```const```, e.g 
```typescript
const MINUTES_IN_HOUR = 60;
```

Reply:
I think we should fail fast in this case. I wanted to check several browsers before make a decision.

Reply:
We agreed to throw an exception here